update.go

```diff
@@ -338,8 +339,22 @@ func (u *Update) FromStream(updateWith io.Reader) (err error, errRecover error)
 			return
 		}
 	case PATCHTYPE_NONE:
-		// no patch to apply, go on through (with bzip2 decoding).
-		newBytes, err = ioutil.ReadAll(bzip2.NewReader(updateWith))
+		// no patch to apply, go on through
+		var fileHeader []byte
+		bufBytes := bufio.NewReader(updateWith)
+		fileHeader, err = bufBytes.Peek(2)
+		if err != nil {
+			return
+		}
+
+		if bytes.Equal([]byte{0x42, 0x5a}, fileHeader) {
```

Comment:
A more readable and safer way would be to check if `ioutil.ReadAll()` triggers `bzip2.StructuralError`, and retry with plain reader if so.